--- poldek-0.30/Makefile.am~	2012-01-26 11:30:31.000000000 +0200
+++ poldek-0.30/Makefile.am	2012-01-26 11:31:12.502250375 +0200
@@ -21,7 +21,7 @@
 
 libpoldek_la_LDFLAGS = -version-info $(libVERINFO)
 libpoldek_la_SOURCES = \
-	  compiler.h	\	
+	  compiler.h	\
 	  log.c log.h	   \
 	  poldek_term.c poldek_term.h	\
 	  minfo.c			    \
@@ -74,10 +74,11 @@
 
 LDADD_              = @INTLLIBS@
 
-pkglib_SCRIPTS      = zlib-in-rpm.sh vfjuggle vfcompr vfsmb poldekuser-setup.sh
+vfscriptsdir = $(libdir)/$(PACKAGE_NAME)
+vfscripts_SCRIPTS = zlib-in-rpm.sh vfjuggle vfcompr vfsmb poldekuser-setup.sh
 
 EXTRA_DIST = \
-	     $(pkglib_SCRIPTS) \
+	     $(vfscripts_SCRIPTS) \
 	     Makefile.extra \
 	     poldek.spec \
 	     package-set.sample vfjugglerc.sample \
