--- poldek-0.30.0/cli/shell.c.orig	2014-02-15 09:09:58.005201258 +0100
+++ poldek-0.30.0/cli/shell.c	2014-02-15 09:10:39.141637805 +0100
@@ -425,7 +425,7 @@ static char **poldek_completion(const ch
         matches = rl_completion_matches(text, command_options_generator);
 
     } else {
-        rl_completer_word_break_characters = " \t\n\"\\'`@$><=;|&{(";
+        rl_completer_word_break_characters = " \t\n\"\\'`$><=;|&{(";
         
         switch (sh_ctx.completion_ctx) {
             case COMPLETITION_CTX_DIRNAME:
@@ -434,7 +434,7 @@ static char **poldek_completion(const ch
                 
             case COMPLETITION_CTX_WHAT_PROVIDES:
             case COMPLETITION_CTX_WHAT_REQUIRES:
-                rl_completer_word_break_characters = " \t\n\"\\'`@$><=;|&{";
+                rl_completer_word_break_characters = " \t\n\"\\'`$><=;|&{";
                 matches = rl_completion_matches(text, deps_generator);
                 break;
                 
