diff --git a/pkgset-dep.c b/pkgset-dep.c
index ddaf74e..f035cd9 100644
--- a/pkgset-dep.c
+++ b/pkgset-dep.c
@@ -293,6 +293,8 @@ tn_array *get_conflicted(int indent, struct pkgset *ps,
     const struct capreq_idx_ent *ent;
     const char *cnflname = capreq_name(cnfl);
 
+    pkgset__index_caps(ps);
+
     if ((ent = capreq_idx_lookup(&ps->cap_idx, cnflname, capreq_name_len(cnfl)))) {
         struct pkg **suspkgs = (struct pkg **)ent->pkgs;
         int nmatch = 0;
