--- poldek-0.42.2/pm/rpmorg/install.c.orig	2020-01-25 22:59:59.000000000 +0100
+++ poldek-0.42.2/pm/rpmorg/install.c	2024-11-26 23:59:38.132248185 +0100
@@ -77,8 +77,8 @@
 
 static void *install_cb(const void *h __attribute__((unused)),
                         const enum rpmCallbackType_e op,
-                        const long long unsigned int amount,
-                        const long long unsigned int total,
+                        const long unsigned int amount,
+                        const long unsigned int total,
                         const void *pkgpath,
                         void *data __attribute__((unused)))
 {
