diff -ur poldek-0.42.2/cli/cmd_chain.c poldek-0.42.2-booldeps/cli/cmd_chain.c
--- poldek-0.42.2/cli/cmd_chain.c	2020-01-25 22:59:59.000000000 +0100
+++ poldek-0.42.2-booldeps/cli/cmd_chain.c	2022-05-19 16:33:21.159335140 +0200
@@ -29,7 +29,6 @@
 #include "compiler.h"
 #include "i18n.h"
 #include "poldek_util.h"
-#define ENABLE_TRACE 0
 #include "log.h"
 #include "cli.h"
 #include "cmd_chain.h"
diff -ur poldek-0.42.2/conf.c poldek-0.42.2-booldeps/conf.c
--- poldek-0.42.2/conf.c	2020-03-10 11:19:28.000000000 +0100
+++ poldek-0.42.2-booldeps/conf.c	2022-05-19 16:32:35.185649828 +0200
@@ -34,7 +34,6 @@
 #include <trurl/nstream.h>
 #include <vfile/vfile.h>
 
-#define ENABLE_TRACE 0
 #include "compiler.h"
 #include "i18n.h"
 #include "log.h"
diff -ur poldek-0.42.2/depdirs.c poldek-0.42.2-booldeps/depdirs.c
--- poldek-0.42.2/depdirs.c	2020-01-25 22:59:59.000000000 +0100
+++ poldek-0.42.2-booldeps/depdirs.c	2022-05-19 16:32:32.922297335 +0200
@@ -22,7 +22,6 @@
 #include <trurl/nassert.h>
 #include <trurl/nmalloc.h>
 
-#define ENABLE_TRACE 0
 #include "compiler.h"
 #include "log.h"
 #include "i18n.h"
diff -ur poldek-0.42.2/install3/ictx.h poldek-0.42.2-booldeps/install3/ictx.h
--- poldek-0.42.2/install3/ictx.h	2020-04-04 21:01:09.000000000 +0200
+++ poldek-0.42.2-booldeps/install3/ictx.h	2022-05-19 16:32:43.675720130 +0200
@@ -17,7 +17,6 @@
 #include <trurl/trurl.h>
 #include <sigint/sigint.h>
 
-#define ENABLE_TRACE 0
 #include "i18n.h"
 #include "log.h"
 #include "misc.h"
diff -ur poldek-0.42.2/install3/iset.c poldek-0.42.2-booldeps/install3/iset.c
--- poldek-0.42.2/install3/iset.c	2020-01-25 22:59:59.000000000 +0100
+++ poldek-0.42.2-booldeps/install3/iset.c	2022-05-19 16:32:41.449035264 +0200
@@ -14,8 +14,6 @@
 # include "config.h"
 #endif
 
-#define ENABLE_TRACE 0
-
 #include <stdlib.h>
 #include <string.h>
 #include <sys/param.h>          /* for PATH_MAX */
diff -ur poldek-0.42.2/pkg.c poldek-0.42.2-booldeps/pkg.c
--- poldek-0.42.2/pkg.c	2022-05-20 16:21:07.639370807 +0200
+++ poldek-0.42.2-booldeps/pkg.c	2022-05-19 16:49:06.084231070 +0200
@@ -27,7 +27,6 @@
 #include <trurl/nstr.h>
 #include <trurl/nassert.h>
 
-#define ENABLE_TRACE 0
 #include "compiler.h"
 #include "i18n.h"
 #include "log.h"
@@ -683,7 +682,7 @@
             }
 
             if (cap_xmatch_req(cap, req, flags)) {
-                DBGF("  cap[%d] %s -> match\n", i, capreq_snprintf_s(cap));
+                DBGF("  cap[%d] %s -> match (pkg: %s)\n", i, capreq_snprintf_s(cap), pkg->name);
                 return 1;
             }
 
diff -ur poldek-0.42.2/pkgcmp.c poldek-0.42.2-booldeps/pkgcmp.c
--- poldek-0.42.2/pkgcmp.c	2020-03-16 14:27:06.000000000 +0100
+++ poldek-0.42.2-booldeps/pkgcmp.c	2022-05-19 16:34:31.123077977 +0200
@@ -22,7 +22,6 @@
 #include <trurl/nstr.h>
 #include <trurl/nassert.h>
 
-#define ENABLE_TRACE 0
 #include "compiler.h"
 #include "i18n.h"
 #include "log.h"
@@ -360,9 +359,9 @@
 
 #if ENABLE_TRACE
     if (pkg_cmp_name_evr_rev(p1, p2) == 0)
-        logn(LOGNOTICE, "uniq %s: keep %s (score %d), removed %s (score %d)",
+        logn(LOGNOTICE, "uniq %s: keep %s (score %d), removed %s",
              pkg_snprintf_s(p1), pkg_arch(p1), pkg_arch_score(p1),
-             pkg_arch(p2), pkg_archscore(p2));
+             pkg_arch(p2));
 #endif
     rc = pkg_cmp_name_evr_rev(p1, p2);
 
diff -ur poldek-0.42.2/pkgdir/pkgdir.c poldek-0.42.2-booldeps/pkgdir/pkgdir.c
--- poldek-0.42.2/pkgdir/pkgdir.c	2020-03-26 10:22:47.000000000 +0100
+++ poldek-0.42.2-booldeps/pkgdir/pkgdir.c	2022-05-19 16:33:01.562526953 +0200
@@ -35,7 +35,6 @@
 
 #include <vfile/vfile.h>
 
-#define ENABLE_TRACE 0
 #include "compiler.h"
 #include "i18n.h"
 #include "log.h"
diff -ur poldek-0.42.2/pkgdir/source.c poldek-0.42.2-booldeps/pkgdir/source.c
--- poldek-0.42.2/pkgdir/source.c	2020-03-16 15:23:52.000000000 +0100
+++ poldek-0.42.2-booldeps/pkgdir/source.c	2022-05-19 16:32:57.605830198 +0200
@@ -29,8 +29,6 @@
 
 #include <vfile/vfile.h>
 
-#define ENABLE_TRACE 0
-
 #include "compiler.h"
 #include "pkgdir.h"
 #include "pkgdir_intern.h"
diff -ur poldek-0.42.2/pkgdir/source_mkidx.c poldek-0.42.2-booldeps/pkgdir/source_mkidx.c
--- poldek-0.42.2/pkgdir/source_mkidx.c	2020-03-25 21:14:48.000000000 +0100
+++ poldek-0.42.2-booldeps/pkgdir/source_mkidx.c	2022-05-19 16:32:55.049143819 +0200
@@ -29,8 +29,6 @@
 
 #include <vfile/vfile.h>
 
-#define ENABLE_TRACE 0
-
 #include "compiler.h"
 #include "pkgdir.h"
 #include "pkgdir_intern.h"
diff -ur poldek-0.42.2/pkgiter.c poldek-0.42.2-booldeps/pkgiter.c
--- poldek-0.42.2/pkgiter.c	2020-04-04 21:01:09.000000000 +0200
+++ poldek-0.42.2-booldeps/pkgiter.c	2022-05-19 16:33:05.289221486 +0200
@@ -24,7 +24,6 @@
 #include <trurl/nstr.h>
 #include <trurl/nassert.h>
 
-#define ENABLE_TRACE 0
 #include "compiler.h"
 #include "i18n.h"
 #include "log.h"
diff -ur poldek-0.42.2/pkgroup.c poldek-0.42.2-booldeps/pkgroup.c
--- poldek-0.42.2/pkgroup.c	2020-01-25 22:59:59.000000000 +0100
+++ poldek-0.42.2-booldeps/pkgroup.c	2022-05-19 16:33:19.022653641 +0200
@@ -21,8 +21,6 @@
 
 #include <trurl/trurl.h>
 
-#define ENABLE_TRACE 0
-
 #include "compiler.h"
 #include "i18n.h"
 #include "log.h"
diff -ur poldek-0.42.2/pm/pkgdb.c poldek-0.42.2-booldeps/pm/pkgdb.c
--- poldek-0.42.2/pm/pkgdb.c	2020-04-05 21:03:29.000000000 +0200
+++ poldek-0.42.2-booldeps/pm/pkgdb.c	2022-05-19 18:03:20.192624394 +0200
@@ -409,6 +409,7 @@
     struct pkg  pkg;
     int         rc;
 
+    DBGF("header_cap_match_req\n");
     rc = 0;
     memset(&pkg, 0, sizeof(pkg));
     pkg.caps = capreq_arr_new(0);
@@ -456,7 +457,7 @@
     int is_file;
 
     is_file = (*capreq_name(req) == '/' ? 1 : 0);
-    //tracef(0, "%s %d", capreq_snprintf_s(req), n_array_size(exclude));
+    tracef(0, "%s %d", capreq_snprintf_s(req), n_array_size(exclude));
 
     if (!is_file && db_match(db, PMTAG_NAME, req, exclude, ma_flags))
         return 1;
diff -ur poldek-0.42.2/pm/rpm/rpmdb_it.c poldek-0.42.2-booldeps/pm/rpm/rpmdb_it.c
--- poldek-0.42.2/pm/rpm/rpmdb_it.c	2020-01-25 22:59:59.000000000 +0100
+++ poldek-0.42.2-booldeps/pm/rpm/rpmdb_it.c	2022-05-19 16:32:46.145740124 +0200
@@ -14,7 +14,6 @@
 # include "config.h"
 #endif
 
-#define ENABLE_TRACE 0
 #include "i18n.h"
 #include "misc.h"
 #include "log.h"
diff -ur poldek-0.42.2/pm/rpmorg/rpmdb_it.c poldek-0.42.2-booldeps/pm/rpmorg/rpmdb_it.c
--- poldek-0.42.2/pm/rpmorg/rpmdb_it.c	2020-01-25 22:59:59.000000000 +0100
+++ poldek-0.42.2-booldeps/pm/rpmorg/rpmdb_it.c	2022-05-19 16:32:49.022429816 +0200
@@ -14,7 +14,6 @@
 # include "config.h"
 #endif
 
-#define ENABLE_TRACE 0
 #include "i18n.h"
 #include "misc.h"
 #include "log.h"
diff -ur poldek-0.42.2/pm/rpmorg/rpm_pkg_ld.c poldek-0.42.2-booldeps/pm/rpmorg/rpm_pkg_ld.c
--- poldek-0.42.2/pm/rpmorg/rpm_pkg_ld.c	2022-05-20 16:21:07.646037483 +0200
+++ poldek-0.42.2-booldeps/pm/rpmorg/rpm_pkg_ld.c	2022-05-20 15:27:51.747648780 +0200
@@ -218,10 +218,10 @@
             goto l_end;
 
         } else {
-            msg(5, "%s%s: %s\n",
+            DBGF("%s%s: %s, %s\n",
                 cr->cr_flags & CAPREQ_PREREQ ?
                 (pmcap_tag == PMCAP_OBSL ? "obsl" : "pre" ):"",
-                tgs->label, capreq_snprintf_s(cr));
+                tgs->label, capreq_snprintf_s(cr), name);
             n_array_push(arr, cr);
         }
     }
diff -ur poldek-0.42.2/tndb/write.c poldek-0.42.2-booldeps/tndb/write.c
--- poldek-0.42.2/tndb/write.c	2020-01-25 23:27:30.000000000 +0100
+++ poldek-0.42.2-booldeps/tndb/write.c	2022-05-19 16:33:15.672630089 +0200
@@ -28,7 +28,6 @@
 #include <trurl/narray.h>
 #include <trurl/nassert.h>
 
-#define ENABLE_TRACE 0
 #include "compiler.h"
 #include "tndb_int.h"
 #include "tndb.h"
diff -ur poldek-0.42.2/uninstall.c poldek-0.42.2-booldeps/uninstall.c
--- poldek-0.42.2/uninstall.c	2020-01-25 22:59:59.000000000 +0100
+++ poldek-0.42.2-booldeps/uninstall.c	2022-05-19 16:32:52.129121034 +0200
@@ -27,7 +27,6 @@
 #include <trurl/trurl.h>
 
 #include "sigint/sigint.h"
-#define ENABLE_TRACE 0
 #include "compiler.h"
 #include "i18n.h"
 #include "log.h"
diff -ur poldek-0.42.2/upgrade-dist.c poldek-0.42.2-booldeps/upgrade-dist.c
--- poldek-0.42.2/upgrade-dist.c	2020-01-25 22:59:59.000000000 +0100
+++ poldek-0.42.2-booldeps/upgrade-dist.c	2022-05-19 16:33:23.476017725 +0200
@@ -29,7 +29,6 @@
 
 #include <sigint/sigint.h>
 
-#define ENABLE_TRACE 0
 #include "compiler.h"
 #include "i18n.h"
 #include "log.h"
diff -ur poldek-0.42.2/vfile/vopen3.c poldek-0.42.2-booldeps/vfile/vopen3.c
--- poldek-0.42.2/vfile/vopen3.c	2020-01-25 22:59:59.000000000 +0100
+++ poldek-0.42.2-booldeps/vfile/vopen3.c	2022-05-19 16:32:37.749004646 +0200
@@ -50,7 +50,6 @@
 
 #include <trurl/nassert.h>
 #include <trurl/nmalloc.h>
-#define ENABLE_TRACE 0
 #include "compiler.h"
 #include "i18n.h"
 #include "log.h"                /* for DBGF */
