--- poldek-0.30.0/configure.ac~	2013-03-09 13:14:38.000000000 +0200
+++ poldek-0.30.0/configure.ac	2013-05-07 11:18:01.889839519 +0300
@@ -574,7 +574,10 @@
 AC_SUBST(SYSCONFDIR)
 
 
-PKGLIBDIR="`echo $libdir/$PACKAGE`"
+AC_ARG_WITH(pkglibdir,
+[  --with-pkglibdir	program executables],
+PKGLIBDIR=$withval, PKGLIBDIR="`echo $libdir/$PACKAGE`")
+
 while echo ${PKGLIBDIR} | grep '\$' >/dev/null; do
         PKGLIBDIR=$(eval echo ${PKGLIBDIR});
 done
--- poldek-0.30.0/Makefile.am~	2013-03-05 14:05:24.000000000 +0200
+++ poldek-0.30.0/Makefile.am	2013-05-07 11:47:28.936268884 +0300
@@ -78,7 +78,7 @@
 
 LDADD_              = @INTLLIBS@
 
-vfscriptsdir = $(libdir)/$(PACKAGE_NAME)
+vfscriptsdir = $(PKGLIBDIR)
 vfscripts_SCRIPTS = zlib-in-rpm.sh vfjuggle vfcompr vfsmb poldekuser-setup.sh
 
 EXTRA_DIST = \
