diff -ruN poldek-0.30-cvs20080604.13./install3/iset.c poldek-0.30-cvs20080604.13/install3/iset.c
--- poldek-0.30-cvs20080604.13./install3/iset.c	2008-02-03 20:07:42.000000000 +0100
+++ poldek-0.30-cvs20080604.13/install3/iset.c	2008-07-17 23:13:25.449260760 +0200
@@ -47,7 +47,7 @@
     pkg_set_mf(iset->pms, pkg, mflag);
 }
 
-inline int iset_ismarkedf(struct iset *iset, const struct pkg *pkg,
+int iset_ismarkedf(struct iset *iset, const struct pkg *pkg,
                           unsigned mflag)
 {
     return pkg_isset_mf(iset->pms, pkg, mflag);
