--- configure.in.old	2010-06-03 04:15:33.000000000 +0200
+++ configure.in	2010-06-03 05:20:29.000000000 +0200
@@ -421,7 +421,7 @@
 if test -n "$TRURL_DIR"; then
 	AC_MSG_CHECKING(for local trurlib copy in $TRURL_DIR)
 	AC_MSG_RESULT([yes])
-	TRURL_INCLUDE="-I\$(top_srcdir)/$TRURL_DIR/include"
+	TRURL_INCLUDE="-I\$(top_srcdir)/$TRURL_DIR/include -I./$TRURL_DIR/include"
 	TRURL_LDFLAG="-L\$(top_srcdir)/$TRURL_DIR/"
 	LDFLAGS="$LDFLAGS $TRURL_LDFLAG"
 	CPPFLAGS="$CPPFLAGS $TRURL_INCLUDE"
