--- install3/misc.c~	2008-08-18 11:26:41.000000000 +0200
+++ install3/misc.c	2008-11-19 15:47:48.471454682 +0100
@@ -365,7 +365,7 @@
 
             /* possible when the same package exists in both available
                and already installed set */
-            if (i3_is_marked_for_removal(ictx, suspkg)) {
+            if (i3_is_marked_for_removal(ictx, suspkg) && !i3_is_marked(ictx, suspkg)) {
                 pkg_free(suspkg);
                 continue;
             }
