diff -urN poldek-0.19.0-cvs20050613.22.org/tndb/tndb.c poldek-0.19.0-cvs20050613.22/tndb/tndb.c
--- poldek-0.19.0-cvs20050613.22.org/tndb/tndb.c	2005-05-12 20:21:53.000000000 +0200
+++ poldek-0.19.0-cvs20050613.22/tndb/tndb.c	2005-09-11 19:57:11.000000000 +0200
@@ -295,7 +295,7 @@
     if (write)
         return n_stream_write_uint32(st, v);
     else
-        tndb_sign_update(&hdr->sign, &v, sizeof(v));
+	tndb_sign_update_int32(&hdr->sign, v);
     
     return 1;
 }
