diff -urbB poldek-0.19.0-cvs20050921.00.org/tndb/read.c poldek-0.19.0-cvs20050921.00/tndb/read.c
--- poldek-0.19.0-cvs20050921.00.org/tndb/read.c	2005-09-11 18:28:10.000000000 +0200
+++ poldek-0.19.0-cvs20050921.00/tndb/read.c	2005-09-22 20:02:31.885659750 +0200
@@ -288,7 +288,7 @@
     tndb_sign_final(&hdr->sign);
     n = 0;
 
-    printf("Xmd_compute =\n (%s)\n (%s)\n %d\n",
+    DBGF("Xmd_compute =\n (%s)\n (%s)\n %d\n",
            n_strdup(tndb_bin2hex_s(hdr->sign.md, sizeof(sign.md))),
            n_strdup((char*)tndb_bin2hex_s(sign.md, sizeof(sign.md))),
            memcmp(sign.md, hdr->sign.md, sizeof(sign.md)));
