Index: install3/requirements.c
===================================================================
RCS file: /cvsroot/poldek/poldek/install3/requirements.c,v
retrieving revision 1.6
diff -u -r1.6 requirements.c
--- install3/requirements.c	24 Feb 2008 22:10:41 -0000	1.6
+++ install3/requirements.c	2 Mar 2008 23:03:26 -0000
@@ -430,7 +430,6 @@
         
         i3pkg->flags &= ~I3PKG_CROSSROAD;
         if (candidates) {
-            n_assert(n_array_size(candidates) > 1);
             if (number_of_non_blacks(ictx, candidates) > 1) {
                 /* mark current package as crossroad and propagate mark down */
                 i3pkg->flags |= I3PKG_CROSSROAD; 
