diff -urN poldek-0.18.1.org/vfile/vhttp/http.c poldek-0.18.1/vfile/vhttp/http.c
--- poldek-0.18.1.org/vfile/vhttp/http.c	2003-12-24 11:53:54.098439232 +0100
+++ poldek-0.18.1/vfile/vhttp/http.c	2003-12-24 12:14:51.627265792 +0100
@@ -210,16 +210,6 @@
         vhttp_msg_fn("<   %s", &req[n]);
     n += nn;
     
-    nn = n_snprintf(&req[n], sizeof(req) - n, "Pragma: no-cache\r\n");
-    if (*vhttp_verbose > 1)
-        vhttp_msg_fn("<   %s", &req[n]);
-    n += nn;
-    
-    nn = n_snprintf(&req[n], sizeof(req) - n, "Cache-Control: no-cache\r\n");
-    if (*vhttp_verbose > 1)
-        vhttp_msg_fn("<   %s", &req[n]);
-    n += nn;
-    
     if (fmt) {
         va_start(args, fmt);
         nn = n_vsnprintf(&req[n], sizeof(req) - n, fmt, args);
