--- poldek-0.30-cvs20080820.23/pkgu.c~	2009-01-26 20:12:20.700457198 +0200
+++ poldek-0.30-cvs20080820.23/pkgu.c	2009-01-26 20:12:00.937750881 +0200
@@ -576,8 +576,9 @@
     nbuf = n_buf_new(1024);
     for (i=0; i < e_name.cnt; i++) {
         char ts[32];
+        time_t t = times[i];
 
-        strftime(ts, sizeof(ts), "%Y-%m-%d %H:%M:%S", gmtime((time_t*)&times[i]));
+        strftime(ts, sizeof(ts), "%Y-%m-%d %H:%M:%S", gmtime((time_t*)&t));
 
         n_buf_printf(nbuf, "* %s %s\n", ts, names[i]);
         n_buf_printf(nbuf, "%s\n\n", texts[i]);
