diff --git a/install3/requirements.c b/install3/requirements.c
index ea9df61..361c400 100644
--- a/install3/requirements.c
+++ b/install3/requirements.c
@@ -386,7 +392,8 @@ static int process_orphan_req(int indent, struct i3ctx *ictx,
     }
     
     /* try upgrade orphan */
-    if (ts->getop(ts, POLDEK_OP_GREEDY)) {
+    if (ts->getop(ts, POLDEK_OP_GREEDY) && !(tomark && pkg_caps_match_req(tomark, req,
+		POLDEK_MA_PROMOTE_EPOCH || POLDEK_MA_PROMOTE_VERSION))) {
         if (try_to_upgrade_orphan(indent, ictx, pkg, req, tomark))
             goto l_end;
     }
