--- poldek-0.30/cli/shell.c~	2010-11-25 22:30:09.000000000 +0100
+++ poldek-0.30/cli/shell.c	2011-02-06 20:20:23.502000232 +0100
@@ -18,6 +18,13 @@
 # include "config.h"
 #endif
 
+#ifndef __USE_EXTERN_INLINES
+# define __USE_EXTERN_INLINES
+# include <argp.h>
+# undef __USE_EXTERN_INLINES
+#else
+# include <argp.h>
+#endif
 #include <stdio.h>
 #include <stdlib.h>
 #include <stdint.h>
@@ -28,7 +35,6 @@
 #include <sys/errno.h>
 #include <sys/param.h>
 #include <signal.h>
-#include <argp.h>
 #include <fnmatch.h>
 #include <unistd.h>
 
