diff -ruN poldek-0.19.0-cvs20040928./lib_pkgset.c poldek-0.19.0-cvs20040928/lib_pkgset.c
--- poldek-0.19.0-cvs20040928./lib_pkgset.c	2004-09-21 12:13:58.000000000 +0200
+++ poldek-0.19.0-cvs20040928/lib_pkgset.c	2004-09-29 16:20:24.512095120 +0200
@@ -96,7 +96,7 @@
 }
 
 tn_array *poldek_search_avail_packages(struct poldek_ctx *ctx,
-                                       enum pkgset_search_tag tag,
+                                       enum poldek_search_tag tag,
                                        const char *value)
 {
     if (!poldek_load_sources(ctx))
