--- poldek-0.42.2/pm/rpmorg/signature.c.orig	2021-04-17 12:56:41.167478071 +0200
+++ poldek-0.42.2/pm/rpmorg/signature.c	2021-04-17 12:56:12.007632757 +0200
@@ -99,6 +99,9 @@
 
         switch (td.tag) {
             case RPMSIGTAG_RSA:
+                flags |= PKGVERIFY_PGP;
+                flags |= PKGVERIFY_GPG;
+                break;
             case RPMSIGTAG_PGP5:	/* XXX legacy */
             case RPMSIGTAG_PGP:
                 flags |= PKGVERIFY_PGP;
