--- poldek-0.20.1-cvs20070108.22/pm/rpm/rpm.c~	2006-12-18 09:53:02.000000000 +0100
+++ poldek-0.20.1-cvs20070108.22/pm/rpm/rpm.c	2007-05-23 17:16:25.562776091 +0200
@@ -403,7 +403,7 @@
     
 
 #ifndef HAVE__RPMVERCMP /* _rpmvercmp - new in rpm 4.4.7 */       
-extern int rpmvercmp(const char *one, const char *two);
+extern int (*rpmvercmp)(const char *one, const char *two);
 #endif
 
 int pm_rpm_vercmp(const char *one, const char *two)
--- poldek-0.20.1-cvs20070108.22/pm/rpm/pm_rpm.h~	2007-01-08 15:59:16.000000000 +0100
+++ poldek-0.20.1-cvs20070108.22/pm/rpm/pm_rpm.h	2007-05-23 16:58:49.155380566 +0200
@@ -2,6 +2,7 @@
 #define POLDEK_PM_RPM_MODULE_H
 
 #include <rpm/rpmlib.h>
+#include <rpm/rpmevr.h>
 #ifdef HAVE_RPM_4_0_4
 # include <rpm/rpmcli.h>
 #endif
--- poldek-0.20.1-cvs20070108.22/configure.in	2007-05-27 20:05:24.702600301 +0200
+++ poldek-0.20.1-cvs20070108.22-new/configure.in	2007-05-27 20:05:03.038233602 +0200
@@ -302,6 +302,7 @@
 dnl rpm 4.4.7
 dnl  - killed rpmMachineScore and rpmGetVar
 AC_CHECK_FUNCS(rpmMachineScore)
+AC_CHECK_FUNCS(rpmPlatformScore)
 AC_CHECK_FUNCS(rpmGetVar)
 AC_CHECK_FUNCS(rpmioAccess)
 dnl  - new _rpmvercmp()
--- poldek-0.20.1-cvs20070108.22/pm/rpm/misc.c	2007-05-27 20:05:24.705933794 +0200
+++ poldek-0.20.1-cvs20070108.22-new/pm/rpm/misc.c	2007-05-27 20:04:43.153951811 +0200
@@ -182,8 +182,11 @@
     return caps;
 }
 
-
-#ifdef HAVE_RPMMACHINESCORE
+#ifdef HAVE_RPMPLATOFMSCORE
+static int machine_score(int tag, const char *val) {
+    return rpmPlatformScore(val, platpat, nplatpat);
+}
+#elif defined(HAVE_RPMMACHINESCORE)
 static int machine_score(int tag, const char *val)
 {
     int rpmtag = 0, rc;
@@ -249,8 +252,10 @@
     
     if (arch == NULL)
         return 0;
-    
-#ifdef HAVE_RPMMACHINESCORE    
+
+#ifdef HAVE_RPMPLATOFMSCORE
+    rc = rpmPlatformScore(val, platpat, nplatpat);
+#elif defined(HAVE_RPMMACHINESCORE)
     rc = rpmMachineScore(RPM_MACHTABLE_INSTARCH, arch);
 #else
     rc = 9;
