Index: pkgfl.c
===================================================================
RCS file: /cvsroot/installer/poldek/pkgfl.c,v
retrieving revision 1.25
diff -u -r1.25 pkgfl.c
--- pkgfl.c	15 May 2002 15:42:48 -0000	1.25
+++ pkgfl.c	5 Jul 2004 08:18:58 -0000
@@ -467,14 +467,14 @@
     while (ndirs--) {
         struct pkgfl_ent  *flent = NULL;
         char              *dn = NULL;
-        int8_t            dnl = 0;
+        uint8_t            dnl = 0;
         int32_t           nfiles = 0;
         int               loadir;
         
         
         n_buf_it_get_int8(nbufi, &dnl);
         dn = n_buf_it_get(nbufi, dnl);
-
+	n_assert(dn);
         loadir = default_loadir;            
         if (dirs && n_array_bsearch(dirs, dn))
             loadir = include;
