--- poldek-0.18.1/configure.in.wiget	Wed Nov 27 19:56:09 2002
+++ poldek-0.18.1/configure.in	Thu Mar  6 20:11:53 2003
@@ -264,11 +264,11 @@
 fi
 
 
-AC_CHECK_LIB(rpm, rpmReadPackageInfo,[LIBS="$LIBS -lrpm $DBLIB"],
+AC_CHECK_LIB(rpm, rpmReadPackageInfo,[LIBS="-lrpm $DBLIB $LIBS"],
 	AC_CHECK_LIB(rpm, rpmReadPackageHeader,
-		[LIBS="$LIBS -lrpm -lrpmdb $DBLIB"],
+		[LIBS="-lrpm -lrpmdb $DBLIB $LIBS"],
 		AC_CHECK_LIB(rpm, rpmReadPackageFile,
-			[LIBS="$LIBS -lrpm -lrpmdb $DBLIB"
+			[LIBS="-lrpm -lrpmdb -lelf $DBLIB $LIBS"
 			 AC_DEFINE([HAVE_RPM_4_1],1,[])],
 			AC_MSG_ERROR(["rpmlib not found"]),
 			[$DBLIB -lrpmdb]),
@@ -298,7 +298,7 @@
 
 AC_CHECK_FUNCS(openpty,,
 	       AC_CHECK_LIB(util, openpty, 
-	[AC_DEFINE([HAVE_OPENPTY],1,[defined if openpty() is avialable])] [LIBS="$LIBS -lutil"]))
+	[AC_DEFINE([HAVE_OPENPTY],1,[defined if openpty() is avialable])] [LIBS="-lutil $LIBS"]))
 
 AC_CHECK_FUNCS(rpmGetRpmlibProvides,,
 	[AC_MSG_WARN("[poldek will not work fine with rpmlib\(...\) capabilities"])],
@@ -318,7 +318,7 @@
 	AC_CHECK_LIB(trurl, n_hash_size,,
 		[AC_MSG_ERROR(["trurlib >= 0.43.5 not found"])], [$CONF_IN_LDFLAGS])
 else
-	LIBS="$LIBS -ltrurl"
+	LIBS="-ltrurl $LIBS"
 	AC_CONFIG_SUBDIRS(trurlib)
 	NARRAY="\"$srcdir/$TRURL_DIR/include/trurl/narray.h\""
 fi
