diff -ruNp poldek-0.30.1.orig/install3/preinstall.c poldek-0.30.1/install3/preinstall.c
--- poldek-0.30.1.orig/install3/preinstall.c	2014-03-25 23:32:25.000000000 +0100
+++ poldek-0.30.1/install3/preinstall.c	2015-10-21 21:53:10.190934930 +0200
@@ -199,8 +199,10 @@ int i3_pre_ts_install(struct poldek_ts *
         if (!pkg_is_marked(ts->pms, pkg))
             continue;
 
-        if (sigint_reached())
+        if (sigint_reached()) {
+            n_array_cfree(pkgs);
             return -1;
+        }
         
         installable = i3_is_pkg_installable(ts, pkg, 1);
         
