--- poldek-0.30-cvs20080820.23/trurlib/include/trurl/narray.h~	2007-07-12 12:56:59.000000000 +0200
+++ poldek-0.30-cvs20080820.23/trurlib/include/trurl/narray.h	2009-01-30 13:21:51.685019394 +0100
@@ -97,10 +97,10 @@
         ...
 */
 
-int n_array_size(const tn_array *arr);
+size_t n_array_size(const tn_array *arr);
 #ifndef SWIG
 #define n_array_size(arr) n_array_size_inl(arr)
-static inline int n_array_size_inl(const tn_array *arr)
+static inline size_t n_array_size_inl(const tn_array *arr)
 {
     return arr->items;
 }
--- poldek-0.30-cvs20080820.23/trurlib/n_array_nth.c~	2004-02-23 00:00:34.000000000 +0100
+++ poldek-0.30-cvs20080820.23/trurlib/n_array_nth.c	2009-01-30 13:25:53.541072782 +0100
@@ -20,7 +20,7 @@
 }
 
 #undef n_array_size
-int n_array_size(const tn_array *arr)
+size_t n_array_size(const tn_array *arr)
 {
     return arr->items;
 }
