diff -urN poldek-0.21-cvs20070703.00/pm/rpm/rpminstall.c poldek-0.21-cvs20070703.00.new/pm/rpm/rpminstall.c
--- poldek-0.21-cvs20070703.00/pm/rpm/rpminstall.c	2007-07-01 23:04:27.000000000 +0200
+++ poldek-0.21-cvs20070703.00.new/pm/rpm/rpminstall.c	2008-02-20 21:34:24.000000000 +0100
@@ -434,7 +434,7 @@
         int can_ask = poldek_ts_is_interactive_on(ts);
 
         if (nsignerr) {
-            if (!can_ask || poldek__confirm(ts, 0,
+            if (!can_ask || !poldek__confirm(ts, 0,
                                             _("There were signature verification errors. "
                                               "Proceed?")))
                 goto l_err_end;
