--- poldek-0.30-cvs20080519.22/conf/poldek.conf	2008-05-19 22:09:20.000000000 +0200
+++ poldek-0.30-cvs20080519.22.new/conf/poldek.conf	2008-05-19 22:34:49.000000000 +0200
@@ -127,11 +127,10 @@
 #confirm removal = yes
 
 # Let the user select package among equivalents.
-#choose equivalents manually = no
+choose equivalents manually = yes
 
 # Prevent package listed from being upgraded if they are already installed.
-# hold = kernel* foo*.i686 th-test:* *-smp-* th-ready:bar*.x86_64
-#hold = 
+hold = kernel*
 
 # Ignore package list - packages fits given mask will be invisible.
 # ignore = *-smp-* foo*
