diff -ruN poldek-0.19.0-cvs20041011.21./pm/rpm/rpm_pkg_ld.c poldek-0.19.0-cvs20041011.21/pm/rpm/rpm_pkg_ld.c
--- poldek-0.19.0-cvs20041011.21./pm/rpm/rpm_pkg_ld.c	2004-09-28 18:52:44.000000000 +0200
+++ poldek-0.19.0-cvs20041011.21/pm/rpm/rpm_pkg_ld.c	2004-12-10 00:27:20.376154952 +0100
@@ -162,24 +162,23 @@
             if (flag & RPMSENSE_EQUAL) 
                 cr_relflags |= REL_EQ;
 
-                
+            if (crtype == PMCAP_REQ) {
 #ifndef HAVE_RPM_EXTDEPS
-            if (flag & RPMSENSE_PREREQ) {
-                n_assert(crtype == PMCAP_REQ);
-                cr_flags |= CAPREQ_PREREQ | CAPREQ_PREREQ_UN;
+                if (flag & RPMSENSE_PREREQ) {
+                    n_assert(crtype == PMCAP_REQ);
+                    cr_flags |= CAPREQ_PREREQ | CAPREQ_PREREQ_UN;
+                }
             }
 #else
-            if (isLegacyPreReq(flag)) { /* prepared by rpm < 4.0.2  */
-                n_assert(crtype == PMCAP_REQ);
-                cr_flags |= CAPREQ_PREREQ | CAPREQ_PREREQ_UN;
+                if (isLegacyPreReq(flag)) { /* prepared by rpm < 4.0.2  */
+                    cr_flags |= CAPREQ_PREREQ | CAPREQ_PREREQ_UN;
                 
-            } else if (isInstallPreReq(flag)) {
-                n_assert(crtype == PMCAP_REQ);
-                cr_flags |= CAPREQ_PREREQ;
+                } else if (isInstallPreReq(flag)) {
+                    cr_flags |= CAPREQ_PREREQ;
                 
-            } else if (isErasePreReq(flag)) {
-                n_assert(crtype == PMCAP_REQ);
-                cr_flags |= CAPREQ_PREREQ_UN;
+                } else if (isErasePreReq(flag)) {
+                    cr_flags |= CAPREQ_PREREQ_UN;
+		}
             }
 #endif /* HAVE_RPM_EXTDEPS */
         }
