Index: install3/process.c
===================================================================
RCS file: /cvsroot/poldek/poldek/install3/process.c,v
retrieving revision 1.5
diff -u -r1.5 process.c
--- install3/process.c	13 May 2008 16:25:09 -0000	1.5
+++ install3/process.c	7 Jun 2008 07:36:54 -0000
@@ -61,6 +61,9 @@
 {
     int i;
 
+    if (i3_is_hand_marked(ictx, i3pkg->pkg))
+        return;
+
     tracef(indent, "- rollbacking %s", pkg_id(i3pkg->pkg));
     iset_remove(ictx->inset, i3pkg->pkg);
     i3_forget_error(ictx, i3pkg->pkg);
