--- poldek-0.20.1-cvs20070108.22/configure.in.orig	2007-04-16 06:33:41.000000000 +0200
+++ poldek-0.20.1-cvs20070108.22/configure.in	2007-04-16 06:34:34.000000000 +0200
@@ -58,10 +58,7 @@
 AM_CFLAGS="-Wall -W"
 AC_SUBST(AM_CFLAGS)
 
-AM_PATH_CHECK(,[have_check="yes"],
-               [AC_MSG_WARN([check library not found; cannot run unit tests]) 
-	       have_check="no"])
-
+PKG_CHECK_MODULES(CHECK, check >= 0.9.4, have_check=yes, have_check=no)
 
 dnl Process features.
 AC_CHECK_LIB(tinfo, tigetstr,,[
