--- poldek-0.20/conf/poldek.conf~	2006-05-15 16:37:25.044005851 +0300
+++ poldek-0.20/conf/poldek.conf	2006-05-15 16:38:11.555045489 +0300
@@ -135,7 +135,8 @@
 
 # Ignore package list - packages fits given mask will be invisible.
 # ignore   = *-smp-* foo*
-#ignore = 
+# we ignore vserver-packages by default, util-vserver has it's own poldek config.
+ignore = vserver-packages
 
 # Conflicts with already installed packages are treated as resolved 
 # if at least one of multiple-installed package satisfies conflict. 
