diff -urN poldek-0.21-cvs20070703.00.org/misc.c poldek-0.21-cvs20070703.00/misc.c
--- poldek-0.21-cvs20070703.00.org/misc.c	2007-07-02 20:30:48.000000000 +0200
+++ poldek-0.21-cvs20070703.00/misc.c	2007-11-22 18:09:26.899099602 +0100
@@ -526,6 +526,9 @@
 {
     if (strstr(path, "./") == NULL)
         return NULL;
+
+    if (strncmp(path, ".", 1) == 0)
+	return get_current_dir_name();
     
 #ifdef HAVE_CANONICALIZE_FILE_NAME /* have safe GNU ext? */
     return realpath(path, NULL);
