diff -urN poldek-0.18.1.org/rpm.c poldek-0.18.1/rpm.c
--- poldek-0.18.1.org/rpm.c	2003-08-19 01:32:35.000000000 +0200
+++ poldek-0.18.1/rpm.c	2003-08-19 01:32:51.000000000 +0200
@@ -531,7 +531,7 @@
     if (count > 0 && (cmprc || dbrecp)) {
         dbrec = rpmdb_it_get(&it);
         
-        if (cmprc)
+        if (cmprc && dbrec)
             *cmprc = -hdr_pkg_cmp_evr(dbrec->h, pkg);
         
         if (dbrecp) {
