diff -r -u poldek-0.18.3/vfile/vfprogress.c poldek-0.18.3-patrys/vfile/vfprogress.c
--- poldek-0.18.3/vfile/vfprogress.c	2004-04-02 12:32:32.240140304 +0200
+++ poldek-0.18.3-patrys/vfile/vfprogress.c	2004-04-02 13:11:02.475931080 +0200
@@ -121,7 +121,7 @@
         return;
     }
     
-    n_assert(bar->prev_n < 100);
+//    n_assert(bar->prev_n <= bar_width); /* not needed */
 
     if (!bar->is_tty) {
 
