Index: pkgset-order.c
===================================================================
RCS file: /cvsroot/installer/poldek/pkgset-order.c,v
retrieving revision 1.16.4.3
retrieving revision 1.16.4.3.2.1
diff -u -r1.16.4.3 -r1.16.4.3.2.1
--- pkgset-order.c	5 May 2005 19:02:25 -0000	1.16.4.3
+++ pkgset-order.c	16 Jul 2005 08:09:28 -0000	1.16.4.3.2.1
@@ -147,7 +147,8 @@
 
                     } else {
                         int i;
-                        poldek_log(LOGERR, _("PreReq loop: "));
+                        if (verbose >= -1)
+                            poldek_log(LOGERR, _("PreReq loop: "));
                         msg(-1, "_%s", rp->pkg->name);
                         for (i=n_array_size(vs->stack)-1; i >= 0; i--) {
                             struct pkg *p = n_array_nth(vs->stack, i);
Index: vfile/vfile.c
===================================================================
RCS file: /cvsroot/installer/poldek/vfile/vfile.c,v
retrieving revision 1.43.4.5
retrieving revision 1.43.4.5.2.1
diff -u -r1.43.4.5 -r1.43.4.5.2.1
--- vfile/vfile.c	15 May 2005 17:22:22 -0000	1.43.4.5
+++ vfile/vfile.c	16 Jul 2005 08:10:37 -0000	1.43.4.5.2.1
@@ -293,7 +293,7 @@
 { 
     int rc = 0;
 
-    if (vfmode & VFM_RW)
+    if ((vfmode & VFM_RW) && (vfmode & VFM_APPEND) == 0)
         vf_unlink(path);
 
     switch (vf->vf_type) {
@@ -354,7 +354,10 @@
                         free(vf->vf_iobuf);
                         vf->vf_iobuf = NULL;
                     }
-                    fseek(vf->vf_stream, 0, SEEK_SET); /* XXX glibc BUG (?) */
+                    fseek(vf->vf_stream, 0L, SEEK_SET); /* XXX glibc BUG (?) */
+                    if (vfmode & VFM_APPEND)
+	                fseek(vf->vf_stream, 0L, SEEK_END); 
+		    
                 } else
                     vfile_err_fn("fopencookie %s: hgw error\n", CL_URL(path));
 
