--- poldek-0.42.2/doc/manual.xml.orig	2020-01-25 22:59:59.000000000 +0100
+++ poldek-0.42.2/doc/manual.xml	2020-04-13 16:10:47.472359736 +0200
@@ -44,7 +44,7 @@
  <titleabbrev role="texinfo-node">poldek</titleabbrev>
  <abstract role="texinfo-node">
   <para>
-    full-featured frontend to RPM.
+    Full-featured frontend to RPM
   </para>
  </abstract>
 
--- poldek-0.42.2/doc/Makefile.am.orig	2020-01-25 22:59:59.000000000 +0100
+++ poldek-0.42.2/doc/Makefile.am	2020-04-13 16:12:58.478316683 +0200
@@ -81,7 +81,7 @@
 	docbook2X2texi --xinclude $< -o $(@D) || docbook2texi --xinclude $< || true
 	perl -pi -e 's|\@emph\{|\@samp\{|g' poldek.texi
 	makeinfo --no-split --force poldek.texi -o $@
-	perl -pi -e 's/^\* poldek manual: \(poldek\)(.+)$$/* poldek: (poldek)$$1/' $@
+	perl -pi -e 's/^\* poldek manual: \(poldek\)\. *(.+)$$/* poldek: (poldek).\t\t\t$$1/' $@
 	rm -f *.texi
 
 manual.html: manual.xml.xhtml.tmp
