--- poldek-0.20/conf/poldek.conf~	2005-10-28 19:14:08.000000000 +0300
+++ poldek-0.20/conf/poldek.conf	2006-03-10 16:45:19.000000000 +0200
@@ -127,7 +127,7 @@
 #confirm removal = yes
 
 # Let the user select package among equivalents.
-#choose equivalents manually = no
+choose equivalents manually = yes
 
 # Prevent package listed from being upgraded if they are already installed.
 # hold = kernel* dev
