--- poldek-0.18.1/configure.in.already_patched	2003-11-26 17:05:45.000000000 +0100
+++ poldek-0.18.1/configure.in	2003-11-26 17:18:00.000000000 +0100
@@ -241,10 +241,8 @@
 		DBLIB="$DBLIB -ldb1"
 	fi
 else
-        AC_CHECK_LIB(db-4.1, db_create, [DBLIB="-ldb-4.1"], 
-	     [AC_CHECK_LIB(db-4.0, db_create, [DBLIB="-ldb-4.0"], 
-		     [AC_CHECK_LIB(db, db_create, [DBLIB="-ldb"],,[$LIBS])], 
-		     [$LIBS])],, [$LIBS])
+	# Check for "-ldb4" only
+	AC_CHECK_LIB(db, db_create, [DBLIB="-ldb4"],,[$LIBS]) 
 
 	if test "${ENABLE_STATIC}." = "yes."; then 
 		DBLIB="$DBLIB"
