--- poldek-cvs20040323/pkgdir_digest.c.orig	2002-05-15 17:53:22.000000000 +0200
+++ poldek-cvs20040323/pkgdir_digest.c	2004-04-24 23:39:17.927522138 +0200
@@ -197,7 +197,8 @@
 static
 int hdr_digest(FILE *stream, unsigned char *md, int *md_size, EVP_MD_CTX *_ctx)
 {
-    int             line_size = 0, nread, len, endvhdr_found = 0;
+    size_t          line_size = 0;
+    int             nread, len, endvhdr_found = 0;
     unsigned char   buf[256];
     char            *linebuf = NULL;
     EVP_MD_CTX      ctx;
--- poldek-cvs20040323/trurlib/n_array_dump_stats.c.orig	2000-10-29 22:12:21.000000000 +0100
+++ poldek-cvs20040323/trurlib/n_array_dump_stats.c	2004-04-24 23:45:50.159356575 +0200
@@ -5,6 +5,6 @@
 void n_array_dump_stats(const tn_array *arr, const char *name)
 {
 
-    printf("\nArray \"%s\" [memsize, items, start_index] = %d, %d, %d\n",
+    printf("\nArray \"%s\" [memsize, items, start_index] = %zd, %zd, %zd\n",
            name ? name : "", arr->allocated, arr->items, arr->start_index);
 }
--- poldek-cvs20040323/trurlib/n_hash_stats.c.orig	2000-10-29 22:12:21.000000000 +0100
+++ poldek-cvs20040323/trurlib/n_hash_stats.c	2004-04-24 23:47:50.222234706 +0200
@@ -31,7 +31,7 @@
             maxdeep = deep;
     }
 
-    printf("%p %d items, %d emptys, %d collisions, maxdeep %d\n", ht, 
+    printf("%p %zd items, %d emptys, %d collisions, maxdeep %d\n", ht, 
            ht->items, nempts, ncolls, maxdeep);
     return n;
 }
--- poldek-cvs20040323/trurlib/nstr_tokl.c.orig	2003-04-30 17:53:02.000000000 +0200
+++ poldek-cvs20040323/trurlib/nstr_tokl.c	2004-04-24 23:48:43.078415374 +0200
@@ -24,6 +24,7 @@
  */
 #include <ctype.h>
 #include <stdlib.h>
+#include <stdio.h>
 #include <string.h>
 
 #include "nmalloc.h"
--- poldek-cvs20040323/vfile/vhttp/test_vhttp.c.orig	2002-04-08 19:21:37.000000000 +0200
+++ poldek-cvs20040323/vfile/vhttp/test_vhttp.c	2004-04-24 23:52:37.708518571 +0200
@@ -1,4 +1,5 @@
 #include <stdio.h>
+#include <stdlib.h>
 #include "vhttp.h"
 #include "http.h"
 
--- poldek-cvs20040323/pkgdir.c.orig	2004-03-11 00:27:42.000000000 +0100
+++ poldek-cvs20040323/pkgdir.c	2004-04-24 23:57:45.955765417 +0200
@@ -509,7 +509,8 @@
     char            *dn, *bn;
     struct vfile    *vf;
     char            *linebuf = NULL;
-    int             line_size = 0, nread, nerr = 0, rc;
+    size_t          line_size = 0;
+    int             nread, nerr = 0, rc;
     const char      *errmsg_broken_difftoc = _("%s: broken patch list");
     char            current_mdd[PDIGEST_SIZE + 1];
     struct pdigest  pdg_current;
@@ -719,7 +720,8 @@
     struct pkgdir        *pkgdir = NULL;
     struct vfile         *vf;
     char                 *linebuf;
-    int                  line_size, nline;
+    size_t               line_size;
+    int                  nline;
     int                  nerr = 0, nread;
     struct pkgroup_idx   *pkgroups = NULL;
     time_t               ts = 0, ts_orig = 0;
@@ -990,7 +992,7 @@
     struct pkgtags_s   pkgt;
     off_t              offs;
     char               *linebuf;
-    int                line_size;
+    size_t             line_size;
     int                nerr = 0, nread, i;
     struct vfile       *vf;
     int                flag_skip_bastards = 0, flag_fullflist = 0;
--- poldek-cvs20040323/rpm_signature.c.orig	2003-07-16 12:49:28.000000000 +0200
+++ poldek-cvs20040323/rpm_signature.c	2004-04-25 00:04:20.178184651 +0200
@@ -52,7 +52,15 @@
 #include "capreq.h"
 #include "rpmdb_it.h"
 
-
+#ifdef HAVE_RPM_4_2
+/* from internal lib/signature.h, no public prototype (at least in 4.3 snaps) */
+typedef enum sigType_e {
+    RPMSIGTYPE_HEADERSIG= 5     /*!< Header style signature */
+} sigType;
+rpmRC rpmReadSignature(FD_t fd, Header *sighp, sigType sig_type, const char ** msg);
+/* lib/rpmlead.h */
+rpmRC readLead(FD_t fd, /*@out@*/ struct rpmlead *lead);
+#endif
 
 static int rpm_read_signature(FD_t fd, Header *sighp, int sig_type)
 {
