--- poldek-0.21-cvs20070703.00/conf/poldek.conf~	2008-02-14 16:12:15.604671074 +0200
+++ poldek-0.21-cvs20070703.00/conf/poldek.conf	2008-02-14 16:13:12.373118918 +0200
@@ -160,6 +160,9 @@
 # Subtle dependency processing options, they should not be changed 
 # unless you know what you are doing
 
+# RPM 4.4.6 introduces auto dependencies based on package directories. Option controls this feature.
+auto directory dependencies = no
+
 # Automatically install packages required by installed ones.
 #follow = yes
 
--- poldek-0.21-cvs20070703.00/conf_sections.c~	2007-06-30 10:42:33.000000000 +0300
+++ poldek-0.21-cvs20070703.00/conf_sections.c	2008-02-15 05:00:16.039094830 +0200
@@ -205,7 +205,7 @@
              
    { "auto directory dependencies", 
      CONF_TYPE_BOOLEAN3,
-             "auto",
+             "no",
              POLDEK_OP_AUTODIRDEP, { 0 } },
              
    { "follow", 
