Index: rpm_signature.c
===================================================================
RCS file: /cvsroot/installer/poldek/Attic/rpm_signature.c,v
retrieving revision 1.4.4.1
diff -u -r1.4.4.1 rpm_signature.c
--- rpm_signature.c	2 Apr 2003 11:39:17 -0000	1.4.4.1
+++ rpm_signature.c	30 Apr 2003 09:28:08 -0000
@@ -59,7 +59,7 @@
 #ifdef HAVE_RPM_4_2
     return rpmReadSignature(fd, sighp, sig_type, NULL) == 0;
 #elif HAVE_RPM_4_1
-    return rpmReadSignature(fd, sighp, sig_type) == 0;
+    return rpmReadSignature(fd, sighp, sig_type, NULL) == 0;
 #elif HAVE_RPM_4_0
     return rpmReadSignature(fd, sighp, sig_type, NULL) == 0;
 #endif
