diff -urN poldek-0.18.1.org/test_match.c poldek-0.18.1/test_match.c
--- poldek-0.18.1.org/test_match.c	2004-01-09 19:48:04.198189038 +0100
+++ poldek-0.18.1/test_match.c	2004-01-09 19:48:23.112218155 +0100
@@ -47,7 +47,7 @@
             while (relases[j]) {
                 struct capreq *cap;
                 
-                pkg = pkg_new("poldek", epoch, versions[i], relases[j], NULL, NULL, 0, 0, 0);
+                pkg = pkg_new("poldek", epoch, versions[i], relases[j], NULL, NULL, NULL, 0, 0, 0);
                 cap = capreq_new("poldek", epoch, versions[i++], relases[j++], REL_EQ, 0);
                 k = 0;
                 while (rels[k] > 0) {
@@ -80,7 +80,7 @@
     struct capreq *req, *cap;
     int rc;
     
-    pkg = pkg_new("poldek", epoch, ver, rel,  NULL, NULL, 0, 0, 0);
+    pkg = pkg_new("poldek", epoch, ver, rel,  NULL, NULL, NULL, 0, 0, 0);
     cap = capreq_new("poldek", epoch, ver, rel, REL_EQ, 0);
 
     req = capreq_new_evr("poldek", n_strdup(evr), relation, 0);
