--- pkg.h-orig	2009-03-31 12:08:29.238718195 +0200
+++ pkg.h	2009-03-31 12:08:49.843187842 +0200
@@ -106,7 +106,7 @@
     int32_t      itime;        /* date of installation  */
 
     /* private, don't touch */
-    int16_t      _refcnt;
+    uint16_t      _refcnt;
     tn_alloc     *na;
     int16_t      _buf_size;
     char         _buf[0];  /* private, store all string members */
--- pkg.c-orig	2009-03-31 11:37:37.825570800 +0200
+++ pkg.c	2009-03-31 12:05:24.631861551 +0200
@@ -1458,7 +1458,7 @@
              pkg->na ? pkg->na->_refcnt : -1,
              pkg->_refcnt, &pkg->_refcnt);
     }
-    n_assert(pkg->_refcnt < INT16_MAX - 1);
+    n_assert(pkg->_refcnt < UINT16_MAX - 1);
     pkg->_refcnt++;
     return pkg;
 }
