--- poldek-0.19.0-cvs20050613.22/pm/rpm/rpm_pkg_ld.c.orig	2004-12-02 18:40:14.000000000 +0100
+++ poldek-0.19.0-cvs20050613.22/pm/rpm/rpm_pkg_ld.c	2005-07-28 14:54:52.000000000 +0200
@@ -170,10 +170,13 @@
                 }
 #else
                 
+#  if RPMSENSE_PREREQ != RPMSENSE_ANY /* rpm 4.4 drops legacy PreReq support */
                 if (isLegacyPreReq(flag)) /* prepared by rpm < 4.0.2  */
                     cr_flags |= CAPREQ_PREREQ | CAPREQ_PREREQ_UN;
                     
-                else if (isInstallPreReq(flag))
+                else
+#  endif
+		if (isInstallPreReq(flag))
                     cr_flags |= CAPREQ_PREREQ;
 
                 if (isErasePreReq(flag))
