--- vfile/vfreq.c.orig	2008-02-02 22:49:05.000000000 +0100
+++ vfile/vfreq.c	2010-01-15 18:08:33.000000000 +0100
@@ -282,7 +282,7 @@
     else 
         req->uri = n_strdupl(tmp, len);
 
-    len = n_snprintf(tmp, sizeof(tmp), "%s://%s/%s", rreq.proto, rreq.host,
+    len = n_snprintf(tmp, sizeof(tmp), "%s://%s%s", rreq.proto, rreq.host,
                      req->uri);
     req->url = n_strdupl(tmp, len);
     req->port = rreq.port;
