diff -urNbB poldek-0.19.0-cvs20050921.00.org/configure.in poldek-0.19.0-cvs20050921.00/configure.in
--- poldek-0.19.0-cvs20050921.00.org/configure.in	2005-07-17 14:45:53.000000000 +0200
+++ poldek-0.19.0-cvs20050921.00/configure.in	2005-09-21 09:03:23.915451128 +0200
@@ -53,7 +53,6 @@
 AC_CHECK_PROG(POD2MAN, pod2man, pod2man)
 AM_GNU_GETTEXT([external])
 AM_GNU_GETTEXT_VERSION(0.11.5)
-AM_PATH_CHECK()
 
 AM_CFLAGS="-Wall -W"
 AC_SUBST(AM_CFLAGS)
