diff -urp poldek-0.30.1.org/configure.ac poldek-0.30.1/configure.ac
--- poldek-0.30.1.org/configure.ac	2014-03-26 20:23:01.000000000 +0100
+++ poldek-0.30.1/configure.ac	2015-12-16 08:43:07.693420809 +0100
@@ -51,6 +51,8 @@ dnl AC_DEFINE([ENABLE_VFILE_TRURLIO],1,[
 
 AC_DISABLE_STATIC
 
+AC_SYS_LARGEFILE
+
 dnl Checks for programs.
 m4_ifdef([LT_INIT], [LT_INIT], [AC_PROG_LIBTOOL])
 AC_PROG_CC()
Tylko w poldek-0.30.1: configure.ac~
diff -urp poldek-0.30.1.org/tndb/configure.ac poldek-0.30.1/tndb/configure.ac
--- poldek-0.30.1.org/tndb/configure.ac	2014-03-26 20:20:43.000000000 +0100
+++ poldek-0.30.1/tndb/configure.ac	2015-12-16 08:45:36.390318410 +0100
@@ -15,6 +15,8 @@ AM_INIT_AUTOMAKE([1.11 foreign no-dist-g
 AM_MAINTAINER_MODE([enable])
 AM_SILENT_RULES([no])
 
+AC_SYS_LARGEFILE
+
 # Checks for programs.
 m4_ifdef([LT_INIT], [LT_INIT], [AC_PROG_LIBTOOL])
 AC_PROG_CC
diff -urp poldek-0.30.1.org/trurlib/configure.ac poldek-0.30.1/trurlib/configure.ac
--- poldek-0.30.1.org/trurlib/configure.ac	2014-03-26 20:20:45.000000000 +0100
+++ poldek-0.30.1/trurlib/configure.ac	2015-12-16 08:45:25.400054699 +0100
@@ -15,6 +15,8 @@ AM_INIT_AUTOMAKE([1.11 foreign no-dist-g
 AM_MAINTAINER_MODE([enable])
 AM_SILENT_RULES([no])
 
+AC_SYS_LARGEFILE
+
 # Checks for programs.
 m4_ifdef([LT_INIT], [LT_INIT], [AC_PROG_LIBTOOL])
 AC_PROG_CC
--- poldek-0.30.1/vfile/vfff/vfff.h~	2014-03-25 23:32:25.000000000 +0100
+++ poldek-0.30.1/vfile/vfff/vfff.h	2015-12-27 22:03:53.306854171 +0100
@@ -13,6 +13,10 @@
 #ifndef VFFF_VCN_H
 #define VFFF_VCN_H
 
+#ifdef HAVE_CONFIG_H
+# include "config.h"
+#endif
+
 #include <stdarg.h>
 #include <sys/param.h>          /* for PATH_MAX */
 #include <netinet/in.h>
