--- poldek-0.20/conf/poldek.conf	2006-03-10 16:45:19.000000000 +0200
+++ poldek-0.20/conf/poldek.conf	2007-01-31 17:48:31.288247465 +0200
@@ -127,11 +127,11 @@
 #confirm removal = yes
 
 # Let the user select package among equivalents.
-#choose equivalents manually = no
+choose equivalents manually = yes
 
 # Prevent package listed from being upgraded if they are already installed.
 # hold = kernel* dev
-#hold = 
+hold = kernel*
 
 # Ignore package list - packages fits given mask will be invisible.
 # ignore   = *-smp-* foo*
--- poldek-0.20/conf/source.conf~	2005-10-28 19:16:24.000000000 +0300
+++ poldek-0.20/conf/source.conf	2006-05-31 22:45:40.030363503 +0300
@@ -82,4 +82,4 @@
 #hold   = kernel*
 #ignore = kernel*-smp-*
 #dscr   = pl_PL
-  
+
