diff -urN poldek-0.32.2.org/conf/poldek.conf poldek-0.32.2/conf/poldek.conf
--- poldek-0.32.2.org/conf/poldek.conf	2016-02-03 22:02:49.000000000 +0100
+++ poldek-0.32.2/conf/poldek.conf	2017-11-23 09:21:23.347551541 +0100
@@ -15,7 +15,7 @@
 
 # External downloaders configuration, not necessary in most cases, cause
 # to internal HTTP and FTP client.
-#%include fetch.conf
+%include fetch.conf
 
 [global]
 
