diff -Nur poldek-0.20/vfile/Makefile.am poldek-0.20.as_needed-fix/vfile/Makefile.am
--- poldek-0.20/vfile/Makefile.am	2005-05-15 17:16:13.000000000 +0200
+++ poldek-0.20.as_needed-fix/vfile/Makefile.am	2006-08-02 01:46:42.000000000 +0200
@@ -22,7 +22,7 @@
 				      vfffmod.c $(VFCURL_SRCS_) \
 					  vopen3.c vopen3.h vfile_intern.h
 
-libvfile_la_LIBADD = vfff/libvfff.la
+libvfile_la_LIBADD = vfff/libvfff.la -lutil
 
 pkgincludedir = $(includedir)/vfile
 pkginclude_HEADERS = vfile.h p_open.h vopen3.h
