--- poldek-0.30-cvs20080820.23/pkg.c~	2009-01-14 17:33:56.297011561 +0200
+++ poldek-0.30-cvs20080820.23/pkg.c	2009-01-14 17:33:59.170397703 +0200
@@ -1451,15 +1451,13 @@
 
 struct pkg *pkg_link(struct pkg *pkg)
 {
-#if ENABLE_TRACE    
-    if (strcmp(pkg->name, "XX") == 0) {
-        DBGF("%p %s (pdir %s, na->refcnt=%d), refcnt=%d (%p)\n",
+    if (pkg->_refcnt >= INT16_MAX - 1) {
+		trace(1, "%p %s (pdir %s, na->refcnt=%d), refcnt=%d (%p)\n",
              pkg, pkg_snprintf_s(pkg),
              pkg->pkgdir ? pkgdir_idstr(pkg->pkgdir) : "<none>",
              pkg->na ? pkg->na->_refcnt : -1,
              pkg->_refcnt, &pkg->_refcnt);
     }
-#endif
     n_assert(pkg->_refcnt < INT16_MAX - 1);
     pkg->_refcnt++;
     return pkg;
