diff -urN poldek-0.21-cvs20080103.17/conf/poldek.conf poldek-0.21-cvs20080103.17.new/conf/poldek.conf
--- poldek-0.21-cvs20080103.17/conf/poldek.conf	2008-01-03 17:24:33.000000000 +0100
+++ poldek-0.21-cvs20080103.17.new/conf/poldek.conf	2008-01-03 17:38:28.000000000 +0100
@@ -127,11 +127,11 @@
 #confirm removal = yes
 
 # Let the user select package among equivalents.
-#choose equivalents manually = no
+choose equivalents manually = yes
 
 # Prevent package listed from being upgraded if they are already installed.
 # hold = kernel*
-#hold = 
+hold = kernel*
 
 # Ignore package list - packages fits given mask will be invisible.
 # ignore = *-smp-* foo*
diff -urN poldek-0.21-cvs20080103.17/conf/source.conf poldek-0.21-cvs20080103.17.new/conf/source.conf
--- poldek-0.21-cvs20080103.17/conf/source.conf	2008-01-03 17:24:33.000000000 +0100
+++ poldek-0.21-cvs20080103.17.new/conf/source.conf	2008-01-03 17:38:59.000000000 +0100
@@ -85,4 +85,4 @@
 #hold   = kernel*
 #ignore = kernel*-smp-*
 #dscr   = pl_PL
-  
+
