--- poldek-0.30/configure.ac~	2012-11-26 15:41:14.238657356 +0100
+++ poldek-0.30/configure.ac	2012-11-26 15:43:32.665319139 +0100
@@ -308,17 +308,17 @@
 	#	DBLIB="$DBLIB -ldb1"
 	#fi
 else
-    AC_CHECK_LIB(db4, db_create, [DBLIB="-ldb4"],
-        [AC_CHECK_LIB(db-4.1, db_create, [DBLIB="-ldb-4.1"],
-             [AC_CHECK_LIB(db-4.0, db_create, [DBLIB="-ldb-4.0"],
-                     [AC_CHECK_LIB(db, db_create, [DBLIB="-ldb"],,[$LIBS])],
+    AC_CHECK_LIB(db4, db_create, [DBLIB="-ldb4" DBINCLUDE="-I/usr/include/db4"],
+        [AC_CHECK_LIB(db-4.1, db_create, [DBLIB="-ldb-4.1" DBINCLUDE="-I/usr/include/db4.1"],
+             [AC_CHECK_LIB(db-4.0, db_create, [DBLIB="-ldb-4.0" DBINCLUDE="-I/usr/include/db4.0"],
+                     [AC_CHECK_LIB(db-5.2, db_create, [DBLIB="-ldb-5.2" DBINCLUDE="-I/usr/include/db5.2"],,[$LIBS])],
                      [$LIBS])],, [$LIBS])],, [$LIBS])
 
 	if test "${ENABLE_STATIC}." = "yes."; then 
 		DBLIB="$DBLIB"
 	fi
 
-	CPPFLAGS="-I/usr/include/db4 $CPPFLAGS"
+	CPPFLAGS="$DBINCLUDE $CPPFLAGS"
 fi
 
 
