--- pkgcmp.c~	2008-11-19 01:12:49.000000000 +0100
+++ pkgcmp.c	2008-11-19 01:13:36.125519583 +0100
@@ -82,13 +82,10 @@
         return 1;
     
     if (pkg->color && candidate->color)
-        return pkg->color & candidate->color;
+        return (pkg->color & candidate->color) > 0;
 
-    /* same name and candidate without color -> promote candidate */
-    if (pkg->color && pkg_is_kind_of(candidate, pkg)) 
-        return 1;
-
-    return 0;
+    /* either new or old package contains no binary files, let it happen */
+    return 1;
 }
 
 int pkg_eq_capreq(const struct pkg *pkg, const struct capreq *cr) 
