--- install3/install.c.orig	2010-05-02 22:53:38.974101261 +0200
+++ install3/install.c	2010-05-02 22:53:44.817340338 +0200
@@ -289,7 +289,7 @@
 
 int i3_do_poldek_ts_install(struct poldek_ts *ts)
 {
-    int i, nerr = 0, n, is_particle;
+    int i, nerr = 0, n, is_particle, approx_sets = 0;
     struct i3ctx ictx;
     tn_array *pkgs = NULL;
     
@@ -319,6 +319,13 @@
         ts->setop(ts, POLDEK_OP_PARTICLE, 1);
     
     i3ctx_init(&ictx, ts);
+
+    for (i = 0; i < n_array_size(pkgs); i++) {
+        struct pkg *pkg = n_array_nth(pkgs, i);
+        if (!pkg_is_marked_i(ts->pms, pkg)) 
+            continue;
+    	approx_sets++;
+    }
     
     for (i = 0; i < n_array_size(pkgs); i++) {
         struct pkg *pkg = n_array_nth(pkgs, i);
@@ -333,10 +340,10 @@
             if (n > 1) {
                 if (poldek_VERBOSE > 0) {
                     poldek_term_printf_c(PRCOLOR_YELLOW,
-                                         "Installing set #%d\n", n);
+                                         "Installing set #%d of ~%d\n", n, approx_sets);
                     fflush(stdout);
                 }
-                msgn_f(0, "** Installing set #%d\n", n);
+                msgn_f(0, "** Installing set #%d\n of ~%d", n, approx_sets);
             }
             
             n++;
