--- poldek-0.21-cvs20070703.00/pm/rpm/rpm.c~	2007-07-02 00:04:27.000000000 +0300
+++ poldek-0.21-cvs20070703.00/pm/rpm/rpm.c	2008-02-15 05:28:23.107186566 +0200
@@ -221,6 +221,7 @@
         logn(LOGERR, _("%s%s: open rpm database failed"),
              rootdir ? rootdir:"", dbpath ? dbpath : pm->default_dbpath);
         rc = 0;
+		abort(); // XXX maybe re-exec ourselves after poldek binary upgrade?
     }
     
 #if ENABLE_TRACE    
