--- poldek-0.42.2/pm/rpmorg/install.c~	2025-01-03 23:00:37.000000000 +0100
+++ poldek-0.42.2/pm/rpmorg/install.c	2025-01-03 23:07:13.436664378 +0100
@@ -77,8 +77,8 @@
 
 static void *install_cb(const void *h __attribute__((unused)),
                         const enum rpmCallbackType_e op,
-                        const long unsigned int amount,
-                        const long unsigned int total,
+                        const rpm_loff_t amount, 
+                        const rpm_loff_t total,
                         const void *pkgpath,
                         void *data __attribute__((unused)))
 {
