--- poldek-0.42.2/pm/rpmorg/Makefile.am~	2020-01-25 23:03:28.000000000 +0100
+++ poldek-0.42.2/pm/rpmorg/Makefile.am	2020-11-15 10:28:44.040052400 +0100
@@ -8,7 +8,6 @@
 libpm_rpm_la_SOURCES =			\
 	mod.c				\
 	rpmdb_it.c			\
-	rpmdb_depdirs.c			\
 	install.c			\
 	signature.c			\
 	misc.c				\
--- poldek-0.42.2/pm/rpmorg/mod.c~	2020-01-25 22:59:59.000000000 +0100
+++ poldek-0.42.2/pm/rpmorg/mod.c	2020-11-15 10:28:21.190049426 +0100
@@ -26,7 +26,7 @@
     pm_rpm_satisfies,
     pm_rpm_dbpath, 
     pm_rpm_dbmtime,
-    pm_rpm_dbdepdirs,
+    NULL,                       /* pm_rpm_dbdepdirs */
     
 (void *(*)(void *, void *, const char *, const char *, mode_t, tn_hash *))pm_rpm_opendb,
     (void (*)(void *))pm_rpm_closedb,
--- poldek-0.42.2/lib_init.c.orig	2020-03-16 21:23:10.000000000 +0100
+++ poldek-0.42.2/lib_init.c	2021-02-12 19:32:47.579273016 +0100
@@ -1433,8 +1433,9 @@
                     break;
 
                 case POLDEK_OP_AUTODIRDEP:
+#if !defined(HAVE_RPMDSUNAME) || defined(HAVE_RPMORG)
                     v = 0;
-#ifdef HAVE_RPMDSUNAME          /* rpmdsUname - rpm 4.4.6  */
+#else          /* rpmdsUname - rpm 4.4.6  */
                     v = 1;
 #endif
                     break;
--- poldek-0.42.2/conf_sections.c~	2020-01-25 23:10:38.000000000 +0100
+++ poldek-0.42.2/conf_sections.c	2021-02-13 18:32:16.569364907 +0100
@@ -214,9 +214,13 @@
              0, { 0 } },
              
    { "auto directory dependencies", 
+#if defined(HAVE_RPMORG)
+     CONF_TYPE_BOOLEAN3 | CONF_TYPE_F_OBSL, NULL, 0, { 0 } },
+#else
      CONF_TYPE_BOOLEAN3,
              "auto",
              POLDEK_OP_AUTODIRDEP, { 0 } },
+#endif
              
    { "follow", 
      CONF_TYPE_BOOLEAN,
