diff --git a/trurlib/include/trurl/nhash.h b/trurlib/include/trurl/nhash.h
index e03fa9c..3ffe267 100644
--- a/trurlib/include/trurl/nhash.h
+++ b/trurlib/include/trurl/nhash.h
@@ -104,6 +104,7 @@ uint32_t n_hash_compute_index_hash(const tn_hash *ht, uint32_t raw_hash);
 struct trurl_hash_iterator {
     tn_hash *ht;
     int pos;
+    int bpos;
 };
 
 typedef struct trurl_hash_iterator tn_hash_it;
diff --git a/trurlib/n_hash_get.c b/trurlib/n_hash_get.c
index 81c612b..48f0897 100644
--- a/trurlib/n_hash_get.c
+++ b/trurlib/n_hash_get.c
@@ -8,21 +8,39 @@ void n_hash_it_init(tn_hash_it *hi, tn_hash *ht)
 {
     hi->ht = ht;
     hi->pos = 0;
+    hi->bpos = 0;
 }
 
 void *n_hash_it_get(tn_hash_it *hi, const char **key) {
     struct hash_bucket **tbl = hi->ht->table;
     size_t i = hi->pos;
 
-    while (tbl[i] == NULL && i < hi->ht->size)
+    while (i < hi->ht->size && tbl[i] == NULL)
         i++;
 
     if (i >= hi->ht->size)
         return NULL;
 
-    struct hash_bucket *ptr = tbl[i];
+    struct hash_bucket *ptr;
+    int j = 0;
+
+    ptr = tbl[i];
+    while (ptr != NULL) {
+        if (j == hi->bpos)
+            break;
+        ptr = ptr->next;
+        j++;
+    }
+
+    n_assert(ptr);
+
+    if (ptr->next == NULL) {
+        hi->pos = i + 1;
+        hi->bpos = 0;
+    } else {
+        hi->bpos++;
+    }
 
-    hi->pos = i + 1;
     if (key)
         *key = ptr->key;
 
diff --git a/pkgmark.c b/pkgmark.c
index 84ea6f4..9845bce 100644
--- a/pkgmark.c
+++ b/pkgmark.c
@@ -351,8 +351,9 @@ void pkgmark_massset(struct pkgmark_set *pms, int set, uint32_t flag)
         return;
 
     tn_hash_it it;
-    struct pkg_mark *m;
+    n_hash_it_init(&it, pms->ht);
 
+    struct pkg_mark *m;
     while ((m = n_hash_it_get(&it, NULL)) != NULL) {
         if (set)
             m->flags |= flag;
diff --git a/cli/dent.c b/cli/dent.c
index b24a723..f4cd4bb 100644
--- a/cli/dent.c
+++ b/cli/dent.c
@@ -36,13 +36,14 @@ struct pkg_dent *pkg_dent_new(struct poclidek_ctx *cctx, const char *name,
                               struct pkg *pkg, int flags, const char *dirpath)
 {
     struct pkg_dent *ent;
-    int dirpath_at = 0, dirpath_len = 0, len = 0;
+    int name_len = 0, dirpath_at = 0, dirpath_len = 0, len = 0;
 
     if (name) {
         while (*name == '/')
             name++;
 
-        len += strlen(name) + 1;
+        name_len = strlen(name);
+        len += name_len + 1;
         n_assert(flags & PKG_DENT_DIR);
         n_assert(dirpath);
 
@@ -60,7 +61,7 @@ struct pkg_dent *pkg_dent_new(struct poclidek_ctx *cctx, const char *name,
     if (name) {
         char *p;
 
-        memcpy(ent->_buf, name, len);
+        memcpy(ent->_buf, name, name_len + 1);
         ent->name = ent->_buf;
 
         if (dirpath) {
diff --git a/cli/ls.c b/cli/ls.c
index 0027ee0..e2bc01d 100644
--- a/cli/ls.c
+++ b/cli/ls.c
@@ -585,9 +585,10 @@ int do_ls(const tn_array *ents, struct cmdctx *cmdctx, const tn_array *evrs)
             cmdctx_printf(cmdctx, "%-*s %-*s\n",
 			  term_width_div2 + term_width_div2/10 - 1, pkg_name,
 			  (term_width/7), group ? group : "(unset)");
-	}
-        else if (flags & OPT_LS_SOURCERPM) {
-            const char *srcrpm = pkg_srcfilename_s(pkg);
+	} else if (flags & OPT_LS_SOURCERPM) {
+            char buf[512];
+            const char *srcrpm = pkg_srcfilename(pkg, buf, sizeof(buf));
+
             cmdctx_printf(cmdctx, "%-*s %-*s\n",
 			  term_width_div2 + term_width_div2/10 - 1, pkg_name,
 			  (term_width/7), srcrpm ? srcrpm : "(unset)");
