diff -up poldek-0.42.2/pkgfetch.c._orig poldek-0.42.2/pkgfetch.c
diff -up poldek-0.42.2/pm/rpmorg/signature.c._orig poldek-0.42.2/pm/rpmorg/signature.c
--- poldek-0.42.2/pm/rpmorg/signature.c._orig	2021-03-19 15:02:55.589828096 +0100
+++ poldek-0.42.2/pm/rpmorg/signature.c	2021-03-20 18:02:34.286105414 +0100
@@ -177,6 +177,11 @@ static int do_verify_signature(const cha
     if (fdt != NULL && Ferror(fdt) == 0) {
         ts = rpmtsCreate();
         rpmtsSetVfyFlags(ts, vfyflags);
+	if ((flags & (PKGVERIFY_PGP | PKGVERIFY_GPG)) == 0) {
+	    int vfylevel = rpmtsVfyLevel(ts);
+	    vfylevel &= ~RPMSIG_SIGNATURE_TYPE;
+	    rpmtsSetVfyLevel(ts, vfylevel);
+	}
         rc = rpmVerifySignatures(&qva, ts, fdt, n_basenam(path));
         rpmtsFree(ts);
 
