diff --git a/trurlib/include/trurl/nhash.h b/trurlib/include/trurl/nhash.h
index e03fa9c..3ffe267 100644
--- a/trurlib/include/trurl/nhash.h
+++ b/trurlib/include/trurl/nhash.h
@@ -104,6 +104,7 @@ uint32_t n_hash_compute_index_hash(const tn_hash *ht, uint32_t raw_hash);
 struct trurl_hash_iterator {
     tn_hash *ht;
     int pos;
+    int bpos;
 };
 
 typedef struct trurl_hash_iterator tn_hash_it;
diff --git a/trurlib/n_hash_get.c b/trurlib/n_hash_get.c
index 81c612b..48f0897 100644
--- a/trurlib/n_hash_get.c
+++ b/trurlib/n_hash_get.c
@@ -8,21 +8,39 @@ void n_hash_it_init(tn_hash_it *hi, tn_hash *ht)
 {
     hi->ht = ht;
     hi->pos = 0;
+    hi->bpos = 0;
 }
 
 void *n_hash_it_get(tn_hash_it *hi, const char **key) {
     struct hash_bucket **tbl = hi->ht->table;
     size_t i = hi->pos;
 
-    while (tbl[i] == NULL && i < hi->ht->size)
+    while (i < hi->ht->size && tbl[i] == NULL)
         i++;
 
     if (i >= hi->ht->size)
         return NULL;
 
-    struct hash_bucket *ptr = tbl[i];
+    struct hash_bucket *ptr;
+    int j = 0;
+
+    ptr = tbl[i];
+    while (ptr != NULL) {
+        if (j == hi->bpos)
+            break;
+        ptr = ptr->next;
+        j++;
+    }
+
+    n_assert(ptr);
+
+    if (ptr->next == NULL) {
+        hi->pos = i + 1;
+        hi->bpos = 0;
+    } else {
+        hi->bpos++;
+    }
 
-    hi->pos = i + 1;
     if (key)
         *key = ptr->key;
 
diff --git a/pkgmark.c b/pkgmark.c
index 84ea6f4..9845bce 100644
--- a/pkgmark.c
+++ b/pkgmark.c
@@ -351,8 +351,9 @@ void pkgmark_massset(struct pkgmark_set *pms, int set, uint32_t flag)
         return;
 
     tn_hash_it it;
-    struct pkg_mark *m;
+    n_hash_it_init(&it, pms->ht);
 
+    struct pkg_mark *m;
     while ((m = n_hash_it_get(&it, NULL)) != NULL) {
         if (set)
             m->flags |= flag;
diff --git a/cli/dent.c b/cli/dent.c
index b24a723..f4cd4bb 100644
--- a/cli/dent.c
+++ b/cli/dent.c
@@ -36,13 +36,14 @@ struct pkg_dent *pkg_dent_new(struct poclidek_ctx *cctx, const char *name,
                               struct pkg *pkg, int flags, const char *dirpath)
 {
     struct pkg_dent *ent;
-    int dirpath_at = 0, dirpath_len = 0, len = 0;
+    int name_len = 0, dirpath_at = 0, dirpath_len = 0, len = 0;
 
     if (name) {
         while (*name == '/')
             name++;
 
-        len += strlen(name) + 1;
+        name_len = strlen(name);
+        len += name_len + 1;
         n_assert(flags & PKG_DENT_DIR);
         n_assert(dirpath);
 
@@ -60,7 +61,7 @@ struct pkg_dent *pkg_dent_new(struct poclidek_ctx *cctx, const char *name,
     if (name) {
         char *p;
 
-        memcpy(ent->_buf, name, len);
+        memcpy(ent->_buf, name, name_len + 1);
         ent->name = ent->_buf;
 
         if (dirpath) {
diff --git a/cli/ls.c b/cli/ls.c
index 0027ee0..e2bc01d 100644
--- a/cli/ls.c
+++ b/cli/ls.c
@@ -585,9 +585,10 @@ int do_ls(const tn_array *ents, struct cmdctx *cmdctx, const tn_array *evrs)
             cmdctx_printf(cmdctx, "%-*s %-*s\n",
 			  term_width_div2 + term_width_div2/10 - 1, pkg_name,
 			  (term_width/7), group ? group : "(unset)");
-	}
-        else if (flags & OPT_LS_SOURCERPM) {
-            const char *srcrpm = pkg_srcfilename_s(pkg);
+	} else if (flags & OPT_LS_SOURCERPM) {
+            char buf[512];
+            const char *srcrpm = pkg_srcfilename(pkg, buf, sizeof(buf));
+
             cmdctx_printf(cmdctx, "%-*s %-*s\n",
 			  term_width_div2 + term_width_div2/10 - 1, pkg_name,
 			  (term_width/7), srcrpm ? srcrpm : "(unset)");
diff --git a/vfile/vfile.h b/vfile/vfile.h
index 3b55b00..b1612f2 100644
--- a/vfile/vfile.h
+++ b/vfile/vfile.h
@@ -174,11 +174,11 @@ EXPORT int vf_url_as_path(char *buf, size_t size, const char *url);
 
 /* replace password with "x" * len(password) */
 EXPORT const char *vf_url_hidepasswd(char *buf, int size, const char *url);
-#define vf_url_hidepasswd_s(url) vf_url_hidepasswd(alloca(PATH_MAX), PATH_MAX, url)
+#define vf_url_hidepasswd_s(url) vf_url_hidepasswd(alloca(256), 256, url)
 
 /* applies vf_url_hidepasswd() + slim down url string to maxl */
 EXPORT const char *vf_url_slim(char *buf, int size, const char *url, int maxl);
-#define vf_url_slim_s(url, maxl) vf_url_slim(alloca(PATH_MAX), PATH_MAX, url, (maxl) > 40 ? (maxl) : 40)
+#define vf_url_slim_s(url, maxl) vf_url_slim(alloca(256), 256, url, (maxl) > 40 ? (maxl) : 40)
 
 EXPORT char *vf_url_unescape(const char *url);
 
diff --git a/pkg.h b/pkg.h
index 15a68c4..59f3e84 100644
--- a/pkg.h
+++ b/pkg.h
@@ -223,14 +223,14 @@ EXPORT int pkg_has_pkgcnfl(struct pkg *pkg, struct pkg *cpkg);
 
 /* src.rpm */
 EXPORT char *pkg_srcfilename(const struct pkg *pkg, char *buf, size_t size);
-#define pkg_srcfilename_s(pkg) pkg_srcfilename(pkg, alloca(512), 512)
+#define pkg_srcfilename_s(pkg) pkg_srcfilename(pkg, alloca(256), 256)
 
 /* RET %path/%name-%version-%release.%arch.rpm  */
 EXPORT char *pkg_filename(const struct pkg *pkg, char *buf, size_t size);
-#define pkg_filename_s(pkg) pkg_filename(pkg, alloca(512), 512)
+#define pkg_filename_s(pkg) pkg_filename(pkg, alloca(256), 256)
 
 EXPORT char *pkg_path(const struct pkg *pkg, char *buf, size_t size);
-#define pkg_path_s(pkg) pkg_path(pkg, alloca(512), 512)
+#define pkg_path_s(pkg) pkg_path(pkg, alloca(256), 256)
 
 EXPORT char *pkg_localpath(const struct pkg *pkg, char *path, size_t size,
                     const char *cachedir);
@@ -251,13 +251,13 @@ EXPORT int pkg_printf(const struct pkg *pkg, const char *str);
 EXPORT int pkg_snprintf(char *str, size_t size, const struct pkg *pkg);
 EXPORT char *pkg_str(char *str, size_t size, const struct pkg *pkg);
 
-#define pkg_snprintf_s(pkg) pkg_str(alloca(512), 512, pkg)
-#define pkg_snprintf_s0(pkg) pkg_str(alloca(512), 512, pkg)
-#define pkg_snprintf_s1(pkg) pkg_str(alloca(512), 512, pkg)
+#define pkg_snprintf_s(pkg) pkg_str(alloca(256), 256, pkg)
+#define pkg_snprintf_s0(pkg) pkg_str(alloca(256), 256, pkg)
+#define pkg_snprintf_s1(pkg) pkg_str(alloca(256), 256, pkg)
 
 EXPORT int pkg_evr_snprintf(char *str, size_t size, const struct pkg *pkg);
 EXPORT char *pkg_evr_str(char *str, size_t size, const struct pkg *pkg);
-#define pkg_evr_snprintf_s(pkg) pkg_evr_str(alloca(512), 512, pkg)
+#define pkg_evr_snprintf_s(pkg) pkg_evr_str(alloca(256), 256, pkg)
 
 /* must be free()d by pkguinf_free(); see pkgu.h */
 EXPORT struct pkguinf *pkg_uinf(const struct pkg *pkg);
diff --git a/pkgdir/pkg_restore.c b/pkgdir/pkg_restore.c
index e7e9373..8dfb4c9 100644
--- a/pkgdir/pkg_restore.c
+++ b/pkgdir/pkg_restore.c
@@ -202,9 +202,9 @@ struct pkg *pkg_restore_st(tn_stream *st, tn_alloc *na, struct pkg *pkg,
     int                  tag, last_tag, tag_binsize = PKG_STORETAG_SIZENIL;
     const  char          *errmg_double_tag = "%s:%lu: double '%c' tag";
     const  char          *errmg_ldtag = "%s:%lu: load '%c' tag error";
+    int                  load_full_fl = (ldflags & PKGDIR_LD_FULLFLIST);
 
 #if 0
-    printf("FULL %d\n", (ldflags & PKGDIR_LD_FULLFLIST));
     if (depdirs) {
         int i;
         printf("depdirs %p %d\n", depdirs, n_array_size(depdirs));
@@ -387,13 +387,13 @@ struct pkg *pkg_restore_st(tn_stream *st, tn_alloc *na, struct pkg *pkg,
 
             case PKG_STORETAG_FL:
                 pkgt.nodep_files_offs = n_stream_tell(st);
-                //printf("flag_fullflist %d, %p\n", flag_fullflist, depdirs);
-                if ((ldflags & PKGDIR_LD_FULLFLIST) == 0 && depdirs == NULL) {
+                if (!load_full_fl && depdirs == NULL) {
                     pkgfl_skip_st(st);
 
                 } else {
                     tn_tuple *fl;
-                    if (pkgfl_restore_st(na, &fl, st, depdirs, 1) < 0) {
+
+                    if (pkgfl_restore_st(na, &fl, st, load_full_fl ? NULL : depdirs, 1) < 0) {
                         logn(LOGERR, errmg_ldtag, fn, ul_offs, *line);
                         nerr++;
                         goto l_end;
@@ -422,6 +422,10 @@ struct pkg *pkg_restore_st(tn_stream *st, tn_alloc *na, struct pkg *pkg,
 
                         pkgt.pkgfl = ffl;
                     }
+
+                    pkgt.flags |= PKGT_HAS_FILES;
+                    if (load_full_fl)
+                        pkgt.flags |= PKGT_HAS_ALLFILES;
                 }
                 break;
 
